src/quacc/calculators/espresso/core.py

```diff
@@ -0,0 +1,48 @@
+from quacc.calculators.espresso.utils import construct_namelist
+from quacc.calculators.espresso.io import _write, _read
+from quacc.utils.files import load_yaml_calc
+
+# Very simple class that can run calculations ASE style
+# I don't think we even need atoms object here.
```

Comment:
Probably don't need an `Atoms`. Sidenote: I know that the `EspressoProfile` mechanism in ASE lets you define a custom executable (so it can be things other than pw.x) and you might even be able to execute it without an `Atoms` object by doing `calc.run()` or something like that. But I imagine it can't do all the things you're trying to do here. I haven't looked at the new profile interface to QE  much.

Reply:
I thought about tweaking the ase espresso calc to do the job. If you have an idea for passing custom write/read functions in a simple that might be possible without being too ugly

Reply:
Hmm... maybe there's a way. I'll give this a look and let you know if I think of a clean route.

Reply:
When you have time, have a look at the latest commit it should be possible to work around using custom EspressoTemplate and Espresso calc functions. What's left is to code the read/write functions for each binary, if this works then I will be able to clean quite a lot a code. Is this what you were thinking about?

The only binary I can think of that will need additional tweaking is neb.x

Reply:
@tomdemeyere: Yes, something like that was what I had in mind! I wasn't sure the particulars, but indeed I was thinking of something just like what you proposed.